File: C#/0200. Number of Islands.cs
public class Solution {
    public int NumIslands(char[][] grid) {
        int rep=0;
        for(int row=0;row<grid.Length;row++){
            for(int col=0;col<grid[0].Length;col++){
                if(grid[row][col]=='1'){
                    rep++;
                    ReFillIslands(grid,row,col);
                }
            }
        }
        return rep;
    }
    public void ReFillIslands(char[][] grid,int row,int col){
        grid[row][col]='X';
        if(row-1>=0 && grid[row-1][col]=='1')
            ReFillIslands(grid,row-1,col);
        if(col-1>=0 && grid[row][col-1]=='1')
            ReFillIslands(grid,row,col-1);
        if(row+1<grid.Length && grid[row+1][col]=='1')
            ReFillIslands(grid,row+1,col);
        if(col+1<grid[0].Length && grid[row][col+1]=='1')
            ReFillIslands(grid,row,col+1);
    }
}

File: archives/20190519python/0200. Number of Islands.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        rep=0
        for row in range(len(grid)):
            for col in range(len(grid[0])):
                if grid[row][col]=='1':
                    rep+=1
                    self.refillLands(row,col,grid)
        return rep
    
    def refillLands(self,row,col,grid):
        grid[row][col]='0'
        if row-1>=0 and grid[row-1][col]=='1':
            self.refillLands(row-1,col,grid)
        if col-1>=0 and grid[row][col-1]=='1':
            self.refillLands(row,col-1,grid)
        if row+1<len(grid) and grid[row+1][col]=='1':
            self.refillLands(row+1,col,grid)
        if col+1<len(grid[0]) and grid[row][col+1]=='1':
            self.refillLands(row,col+1,grid)
